pkg/apis/core/validation/validation.go

```diff
@@ -1978,7 +1978,7 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 	} else if errs := ValidatePositiveQuantityValue(storageValue, fldPath.Child("resources").Key(string(core.ResourceStorage))); len(errs) > 0 {
 		allErrs = append(allErrs, errs...)
 	} else {
-		allErrs = append(allErrs, ValidateResourceQuantityValue(string(core.ResourceStorage), storageValue, fldPath.Child("resources").Key(string(core.ResourceStorage)))...)
+		allErrs = append(allErrs, ValidateStorageQuantityValue(string(core.ResourceStorage), storageValue, fldPath.Child("resources").Key(string(core.ResourceStorage)))...)
```

Comment:
/hold

standard questions when we want to tighten validation:
* does it prevent a no-op update of existing persisted objects that would fail this new validation? (would block storage migration tasks)
* does it prevent deletion of existing persisted objects with finalizers that would fail this new validation? (update is required to remove finalizers to allow deletion)
* does it prevent controllers from updating status of existing persisted objects that would fail this new validation?

we could consider ratcheting validation:
* apply this tightened validation on create
* apply this tightened validation on update where the old object passed this validation (don't let a valid object become invalid)
* skip this tightened validation on update where the old object fails this validation

c.f. https://github.com/kubernetes/kubernetes/issues/59984

Reply:
I think the only case that the `MilliValute` of `requests.storage` could not by divided by 1000 is user set the value to something like `1234m` which means `1.234` byte which doesn't make any sense in real world. At least the suffix is valid to describe the size, such as 'Ki', 'M', 'Mi' etc., then we are good.

But one can never be too cautious, I agree it is possible to break the API if we tighten the validation. If I try ratcheting validation as you said, is this good to have? If not, I think we could leave the current logic intact.

Reply:
opened https://github.com/kubernetes/kubernetes/pull/94620 to fix the issue with the quota evaluator to only record whole byte values

will separately track warning about this issue in PVC creation